Userland/Applications/TextEditor/TextEditorWidget.cpp

```diff
@@ -685,9 +686,26 @@ void TextEditorWidget::update_html_preview()
     m_page_view->scroll_into_view(current_scroll_pos, true, true);
 }
 
-void TextEditorWidget::update_statusbar_cursor_position()
+void TextEditorWidget::update_statusbar()
 {
     StringBuilder builder;
     builder.appendff("Line: {}, Column: {}", m_editor->cursor().line() + 1, m_editor->cursor().column());
+
+    if (m_editor->has_selection()) {
+        int word_count = 0;
+        bool in_word = false;
+        for (char c : m_editor->selected_text()) {
```

Comment:
Let's put the result of `selected_text()` in a variable since it's a pretty heavy operation that iterates over the selection and creates a new `String` from it:

```cpp
void TextEditorWidget::upate_statusbar()
{
    auto selected_text = m_editor->selected_text();
    ...
```